blue_ai/agents/dqn.py

```diff
@@ -158,7 +158,7 @@ def select_action(self, state):
             )[0]
 
         if self.softmax_temp is not None:
-            return softmax_selection(values.detach().numpy(), self.softmax_temp)
+            return softmax_selection(values.detach().cpu().numpy(), self.softmax_temp)
```

Comment:
I don't think we need this one - things seem to run for me without it. And since this is happening inside the reinforcement learning agent, we don't want to force it onto the CPU.